e2e/testcases/managed_resources_test.go

```diff
@@ -133,18 +150,16 @@ metadata:
 	}
 
 	/* A new test */
```

Comment:
This is a behavior change with the new filtering that I discussed in the design doc under caveats. 

Previously, if the remediator was watching the resource type, because it matched the resource of an object in the source, it would see when a new object was created with the `configsync.gke.io/manager` annotation matching the existing reconciler, then it would delete it, because it knows it doesn't manage it.

But with watch filtering, this only happens if the object ALSO has a matching `applyset.kubernetes.io/part-of` label.

Personally, I don't think users are ever going to care about or want this behavior, because it relies on too many coincidences. 

Like, if you make a ConfigMap and give it the manager annotation of an existing reconciler and there isn't already a ConfigMap in the source for that reconciler, then it wouldn't be watching ConfigMaps, so it wouldn't have deleted it. 

Realistically, this scenario is only going to happen if you export an existing ConfigMap, synced by a reconciler, and then gave it a new name and re-applied it. It could happen, but it's not clear that the user would really want or expect Config Sync to delete it. It might be better to just error. And if that is was happened, then it would also probably have the ApplySet label too.

I don't know if it really make sense to test all the permutations of these edge cases, unless someone runs into them and complains about the behavior as unexpected. Then we would at least have some signal that the behavior we're testing is actually desired. Otherwise, we're just writing regression tests for the implementation, not validating intended behavior.

Reply:
Noted the behavior change. I think it would be good to have a test to demonstrate the new behavior. When a support ticket is filed, we can point to the test to explain the behavior.

A scenario I can think of is a R*Sync is deleted without deletion propagation before watch filtering is supported. All managed resources are kept without the applyset labels. Then a resource is removed from the source repo. 
Then Config Sync is upgraded with the watch filtering support and the RS*sync is re-created. It should be the user intention that the resource should be pruned.

Noted that this is kind of an edge case, but it is introduced by this PR. It would be god to have a test case to cover the new behavior.

Reply:
I added some tests to cover the changed behavior to show that objects won't be deleted if they dont have the right ApplySet ID in the part-of label value. 

I also added a new test at the bottom to validate that the remediator can catch the case where the part-of label is changed or removed.